projects/microphysics/scripts/prognostic_evaluate.py

```diff
@@ -0,0 +1,440 @@
+    level_map: Mapping[str, int] = None,
+):
+
+    if time_idxs is None:
+        time_idxs = {"start": 0, "near_end": len(prognostic.time) - 2}
+
+    if level_map is None:
+        level_map = {"lower": 75, "upper_BL": 60, "upper_atm": 20}
+
+    for name in do_variables:
+        for level, lev_idx in level_map.items():
+            for time, tidx in time_idxs.items():
+                prog = prognostic.isel(time=tidx, z=lev_idx)
+                base = baseline.isel(time=tidx, z=lev_idx)
+                fig = plot_spatial_2panel_with_diff(prog, base, name)
+
+                log_name = f"spatial_comparison/{time}/{level}/{name}"
+                wandb.log({log_name: wandb.Image(plot_to_image(fig))})
+                plt.close(fig)
+
+
+def _selection(
+    times: xr.DataArray, duration: timedelta, window: Optional[timedelta] = None
+):
+
+    """
+    Generate a slice window at the specified duration inferred
+    from the save timestep frequency
+
+    Args:
+        times: Datarray of datetime objects marking the savepoints
+        duration: time elapsed to center the selection window at
+        window: width of the selection window, defaults to +/- one timestep
+    """
+
+    delta = pd.Timedelta((times[1] - times[0]).values)
+    center = duration // delta
+
+    if window is None:
+        increment = 1
+    else:
+        increment = window // delta
+
+    start = max(center - increment, 0)
+    end = center + increment
+
+    return slice(start, end)
+
+
+def log_all_drifts(
+    prog_global_avg: xr.Dataset, base_global_avg: xr.Dataset, do_variables=COMPARE_VARS
+):
+
+    times = prog_global_avg.time
+    for name in do_variables:
+        # assumes time delta
+        drift_sel = {
+            "3hr": _selection(times, timedelta(hours=3)),
+            "1day": _selection(times, timedelta(days=1), window=timedelta(hours=12)),
+            "5day": _selection(times, timedelta(days=5), window=timedelta(hours=12)),
+            "10day": _selection(times, timedelta(days=10), window=timedelta(days=1)),
+        }
+
+        # not quite drift from init but an estimate
+        prog_init = prog_global_avg[name].isel(time=0)
+        base_init = base_global_avg[name].isel(time=0)
+
+        columns = {}
+        columns["drift_def"] = [
+            "baseline_from_init",
+            "prognostic_from_init",
+            "prognostic_from_baseline",
+        ]
+
+        for key, selection in drift_sel.items():
+
+            prog_sel = prog_global_avg[name].isel(time=selection).mean(dim="time")
+            base_sel = base_global_avg[name].isel(time=selection).mean(dim="time")
+
+            columns[key] = [
+                (base_sel - base_init).values.item(),
+                (prog_sel - prog_init).values.item(),
+                (prog_sel - base_sel).values.item(),
+            ]
+
+        df = pd.DataFrame.from_dict(columns)
+        table = wandb.Table(dataframe=df)
+        wandb.log({f"drifts/{name}": table})
```

Comment:
`wandb.Table` is not easily compared accross runs, since it doesn't show up in the wandb table interface and cannot be used in the non-custom chart types. Can you log these as scalars instead like this:

```
wandb.log({"drifts/{name}/{time}": float(drift amount)})
```